src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace SprykerEco\Zed\ArvatoRss\Persistence;
+
+use Generated\Shared\Transfer\ArvatoRssApiCallLogTransfer;
+
+/**
+ * @method \SprykerEco\Zed\ArvatoRss\Persistence\ArvatoRssPersistenceFactory getFactory()
+ */
+class ArvatoRssRepository implements ArvatoRssRepositoryInterface
+{
+    /**
+     * @param string $orderReference
+     * @param string $type
+     *
+     * @return ArvatoRssApiCallLogTransfer|null
+     */
+    public function findApiLogByOrderReferenceAndType(string $orderReference, string $type): ?ArvatoRssApiCallLogTransfer
+    {
+        $arvatoRssApiCallLog = $this->getFactory()->createArvatoRssApiCallLogQuery()
+            ->filterByCallType($type)
+            ->filterByOrderReference($orderReference)
+            ->findOne();
+
+        if ($arvatoRssApiCallLog === null) {
+            return null;
```

Comment:
Would be nice return empty transfer instead.

Reply:
Changed

Reply:
But this is a common practice in core to return null if we didn't find a record.

Reply:
I agree, an empty transfer cloaks the case that this is actually without data. So usually nullable is the cleaner approach isnt it?

Reply:
Okay, core rules are changing rapidly. Looks like my knowledge is outdated.

Reply:
Reverted